### PR TITLE
PR for sourcemap support

### LINKED ISSUE #1: Plugin is not compatible with gulp-sourcemaps
I'm trying to prefix my selectors with this plugin and have sourcemaps generated by [`gulp-sourcemaps`](https://github.com/floridoo/gulp-sourcemaps) but the problem is that your plugin is changing the way my css is written by removing whitespaces.
Someting like this (with `splitOnCommas: true`):

``` css
h1,
h2,
h3 {
 ...
}
```

is rendered out as

``` css
.prefix h1, .prefix h2, .prefix h3 {
   ...
}
```

This in turn messes things up because the sourcemap plugin doesn't account for this change. So can you please add support for `gulp-sourcemaps`? Info on how to do this can be found [here](https://github.com/floridoo/gulp-sourcemaps#plugin-developers-only-how-to-add-source-map-support-to-plugins).

### PR DESCRIPTION
**Pull request by RadValentin**
Fixes #1 

Note: I've added sourcemap support for buffer using the instructions found [here](https://github.com/floridoo/gulp-sourcemaps#plugin-developers-only-how-to-add-source-map-support-to-plugins). For stream I couldn't find a way to make sourcemaps work, none of the other plugins that support `gulp-sourcemaps` allow streaming. Maybe this will change in the future but for now this should suffice.

### CODE REVIEW

**Comment by RadValentin** [comment]
@nixonchris Can you review the changes? 

cc: @skidding @NiGhTTraX @palcu

File: index.js [hunk 1]
```diff
@@ -37,9 +45,30 @@ module.exports = function (fn, options) {
 			file.contents = through;
 
 		} else if (file.isBuffer()) {
-			file.contents = new Buffer(transformSelector.process(file.contents).css);
+
+			if (file.sourceMap) {
+	      options.makeSourceMaps = true;
+	    }
```
Comment (ovidiuch) [inline index.js]: Something's wrong with this indentation

File: index.js [hunk 2]
```diff
@@ -23,9 +25,15 @@ module.exports = function (fn, options) {
 
 	return es.map(function (file, callback) {
 		var through,
-			wait;
+			wait,
+			result;
 
-		if (file.isStream()) {
+		if(file.isNull()) {
```
Comment (ovidiuch) [inline index.js]: Add space `if (`

File: test/main.js [hunk 3]
```diff
@@ -87,4 +88,33 @@ describe('gulp-require-convert', function () {
...
+
+		var srcFile = new gutil.File({
+			path: 'test/fixtures/main.css',
+			cwd: 'test/',
+			base: 'test/fixtures',
+			contents: fs.readFileSync('test/fixtures/main.css')
+		});
+
+		init
+			.pipe(transformSelectors(testTransformation, {
```
Comment (ovidiuch) [inline test/main.js]: What's with this weird spacing here?

Comment (RadValentin) [inline test/main.js]: The original file uses tabs for indentation, I used them as well for consistency.

PS: Github really doesn't like tabs :P

**Comment by RadValentin** [comment]
@skidding I fixed the indentation issues. 

@nixonchris Can you review this PR please?

**Comment by RadValentin** [comment]
@nixonchris if you're still alive somewhere, it's been more than two whole years...👴♿⚰

**Pull request closed without merging**